Add GOVERNANCE.md, answer open gov checklist (#1327)
submariner-io/submariner: GOVERNANCE.md

# Submariner Governance

## Open Governance Checklist

The opengovernance.dev project [provides the following
checklist](https://github.com/opengovernance/opengovernance.dev/blob/master/README.md#checklist) for defining open governance.

Submariner's answers are provided for each point.

This is not meant to define Submariner's governance, but instead to document it clearly. Authoritative sources are cited inline.

### Ownership

* Who owns the copyright on contributed code?
  * All Submariner code is linted to verify the copyright notices specify "Contributors to the Submariner project", per the
    [CNCF recommendations](https://github.com/cncf/foundation/blob/master/copyright-notices.md#copyright-notices).
    **TODO: [Update to this format](https://github.com/submariner-io/submariner/issues/1322)**
* Who owns the domain(s) for the project?

>   * Submariner is a CNCF Sandbox project, so all Submariner domains are owned by CNCF.

Reviewer: We are still waiting for transfer, right?

* Who owns the trademark for the project, is it neutrally owned and governed? Are there open trademark guidelines?

>   * Submariner is a CNCF Sandbox project, so all Submariner trademarks are owned by the CNCF.

Reviewer: Same I guess.

* How can users license the project’s branding? Are there open branding guidelines?
  * Submariner is a CNCF Sandbox project, so please see the CNCF/artwork project's
    [Trademark and Logo Policy](https://github.com/cncf/artwork#trademark-and-logo-policy).
* If the project raises funds, who owns it?

>   * The Submariner Contributors. **TODO: Document this elsewhere.**

Author: Proposed authoritative docs here: https://github.com/submariner-io/submariner-website/pull/520

### Process

* Who makes decisions on how the project performs releases?
  * Project Owners are responsible for defining milestones and releases, per the Submariner
    [Contributor Roles](https://submariner.io/community/contributor-roles/).
* How can the project contributors become committers?
  * In short, by reviewing code and receiving committer/owner approval.
    See the [Committer Requirements](https://submariner.io/community/contributor-roles/) for details.
* How are project committers removed?

>   * By stepping down or majority vote of Project Owners. **TODO: Document this elsewhere.**

Author: Proposed authoritative docs here: https://github.com/submariner-io/submariner-website/pull/520

* If the project raises funds, who decides how this money is spent?
  * The Submariner Contributors. **TODO: Document this elsewhere.**
* Who decides the project roadmap?
  * The [roadmap planning process](https://submariner.io/community/roadmap/#planning-process) is public and open to everyone.
* Who can participate in security disclosure issues?
  * Project Owners or committers who are required for the process, and members of the submariner-security@googlegroups.com email list.

>     **TODO: Document this elsewhere.**

Author: Proposed this as authoritative docs: https://github.com/submariner-io/submariner-website/pull/518

* How transparent are the decision-making processes?
  * Decisions are made in weekly public community meetings or on public GitHub Issues/PRs. All meetings are documented in Submariner's
    [Community Calendar](https://submariner.io/community/getting-help/#community-calendar).
* Who enforces the code of conduct?

>   * The Submariner Project Owners enforce the Code of Conduct ([details](https://submariner.io/community/code-of-conduct)).

Author: Made this more explicit here: submariner-io/submariner-website#520

# Submariner Governance

## Open Governance Checklist

The opengovernance.dev project [provides the following
checklist](https://github.com/opengovernance/opengovernance.dev/blob/master/README.md#checklist) for defining open governance.

Submariner's answers are provided for each point.

This is not meant to define Submariner's governance, but instead to document it clearly. Authoritative sources are cited inline.

### Ownership

* Who owns the copyright on contributed code?

>   * All Submariner code is linted to verify the copyright notices specify "Contributors to the Submariner project", per the

Author: Should maybe link to the linting config: https://github.com/submariner-io/submariner/blob/ed473895ece6dc9be6f9d925a3cfbe441513cf17/.licenserc.json#L22

Author: Done in the latest.

    [CNCF recommendations](https://github.com/cncf/foundation/blob/master/copyright-notices.md#copyright-notices).
* Who owns the domain(s) for the project?
  * Submariner is a CNCF Sandbox project, so all Submariner domains are owned by the CNCF.
* Who owns the trademark for the project, is it neutrally owned and governed? Are there open trademark guidelines?
  * Submariner is a CNCF Sandbox project, so all Submariner trademarks are owned by the CNCF.
* How can users license the project’s branding? Are there open branding guidelines?
  * Submariner is a CNCF Sandbox project, so please see the CNCF/artwork project's
    [Trademark and Logo Policy](https://github.com/cncf/artwork#trademark-and-logo-policy).
* If the project raises funds, who owns it?
  * Submariner is a CNCF Sandbox project, so funds would be owned by the CNCF.

### Process

* Who makes decisions on how the project performs releases?
  * Project Owners are responsible for defining milestones and releases, per the Submariner
    [Owner Responsibilities and Privileges](https://submariner.io/community/contributor-roles/#owner-responsibilities-and-privileges).
* How can the project contributors become committers?

>   * In short, by reviewing code and receiving Committer/Owner approval.

Reviewer 2: ```suggestion
  * In short, by reviewing pull requests and receiving Committer/Owner approval.
```

Documentation PRs also count ;-).

    See the [Committer Requirements](https://submariner.io/community/contributor-roles/#committer-requirements) for details.
* How are project committers removed?
  * Project Owners and Committers can be removed by stepping down or by two thirds vote of Project Owners, per the Submariner
    [Contributor Roles](https://submariner.io/community/contributor-roles).
* If the project raises funds, who decides how this money is spent?

>   * Project Owners are responsible deciding how funds are spent, per the Submariner

Reviewer 2: ```suggestion
  * Project Owners are responsible for deciding how funds are spent, per the Submariner
```

    [Owner Responsibilities and Privileges](https://submariner.io/community/contributor-roles/#owner-responsibilities-and-privileges).
* Who decides the project roadmap?
  * The [roadmap planning process](https://submariner.io/community/roadmap/#planning-process) is public and open to everyone.
* Who can participate in security disclosure issues?
  * Project Owners are responsible for receiving [security disclosures](https://submariner.io/security.txt) and ensuring an adequate
    response, per the Submariner
    [Owner Responsibilities and Privileges](https://submariner.io/community/contributor-roles/#owner-responsibilities-and-privileges).
* How transparent are the decision-making processes?
  * Decisions are made in weekly public community meetings or on public GitHub Issues/PRs. All meetings are documented in Submariner's
    [Community Calendar](https://submariner.io/community/getting-help/#community-calendar).
* Who enforces the code of conduct?

>   * The Submariner Project Owners enforce the [Code of Conduct](https://submariner.io/community/code-of-conduct), per the Submariner.

Author: ```suggestion
  * The Submariner Project Owners enforce the [Code of Conduct](https://submariner.io/community/code-of-conduct), per the Submariner
```

Extra period, sorry.
